### PR TITLE
Bump DScanner & enable for more unused variable checks

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
This removes the following modules from the blacklist:

- dmd.dinterpret
- dmd.dmangle
- dmd.e2ir
- dmd.libmscoff
- dmd.ob
- dmd.semantic2
- dmd.tocvdebug

### CODE REVIEW

File: src/dmd/ob.d [hunk 1]
```diff
@@ -1676,7 +1676,7 @@ void genKill(ref ObState obstate, ObNode* ob)
 
             override void visit(NewExp e)
             {
-                Type tb = e.newtype.toBasetype();
+                // Type tb = e.newtype.toBasetype();
```
Comment (Geod24) [inline src/dmd/ob.d:1679]: No need to comment things out, just remove them.

Comment (MoonlightSentinel) [reply src/dmd/ob.d]: Done

File: src/dmd/ob.d [hunk 2]
```diff
@@ -1514,7 +1511,7 @@ void genKill(ref ObState obstate, ObNode* ob)
                 foreach (i; varStackSave .. obstate.varStack.length)
                 {
                     const vi = obstate.varStack[i];
-                    auto pvs = &ob.gen[vi];
+                    // auto pvs = &ob.gen[vi];
```
Comment (MoonlightSentinel) [inline src/dmd/ob.d:1514]: Keeping this because it's "used" below

File: src/build.d [hunk 3]
```diff
@@ -525,21 +525,13 @@ alias style = makeRule!((builder, rule)
...
-            run([git, "clone", "--depth=15", "--recurse-submodules", "--branch=v0.8.0",
-                "https://github.com/dlang-community/Dscanner", dscannerDir]);
-
-            // TODO: Remove this temporary fix for the invalid error messages in
-            // backend/ptrtab.d when D-Scanner upgrades DParse
-            run([git, "-C", dscannerDir.buildPath("libdparse"), "checkout", "63559db5cc6fa38c01bdda36e09638b5f20fb8e5"]);
-        })
+        .command([
+            // FIXME: Omitted --shallow-submodules because it requires a more recent
+            //        git version which is not available on buildkite
```
Comment (thewilsonator) [inline src/build.d:530]: This comment seems out of date.

Comment (MoonlightSentinel) [reply src/build.d:530]: Last run from BuildKite:

```
[...]
Command: git clone --depth=1 --recurse-submodules --shallow-submodules --branch=v0.9.0-beta.1 https://github.com/dlang-community/Dscanner /var/lib/buildkite-agent/builds/ci-agent-1b8888d4-05cf-49df-8be7-5111d465b14e-1/dlang/dmd/dmd/generated/linux/release/64/dscanner
-------------------------------------------------------------
ERROR: error: unknown option `shallow-submodules'
usage: git clone [<options>] [--] <repo> [<dir>]
[...]
```

Comment (thewilsonator) [reply src/build.d:530]: Ah its not passing.

Comment (MoonlightSentinel) [reply src/build.d:530]: For reference, `--shallow-submodules` is available since `git v2.9.0` which was released in July 2016.

**Review by PetarKirov — APPROVED** [review]

**Pull request merged**